src/connection_string.ts

```diff
@@ -478,12 +485,11 @@ export function parseOptions(
     throw new MongoParseError('Can only specify both of proxy username/password or neither');
   }
 
-  if (
-    urlOptions.get('proxyHost')?.length > 1 ||
-    urlOptions.get('proxyPort')?.length > 1 ||
-    urlOptions.get('proxyUsername')?.length > 1 ||
-    urlOptions.get('proxyPassword')?.length > 1
-  ) {
+  const proxyOptions = ['proxyHost', 'proxyPort', 'proxyUsername', 'proxyPassword'].map(
```

Comment:
Previously, urlOptions was of type `Map<string, any>`, so `.get()` returned a value of the type `any | undefined` which gets simplified to `any`.  So no type errors.

When I added stricter typing to the `urlOptions` object, `.get()` now returns `any[] | undefined`, Typescript complains that `.get(...)?.length` could be undefined.